linkage-monitor/src/main/java/com/google/cloud/tools/dependencies/linkagemonitor/LinkageMonitor.java

```diff
@@ -89,12 +90,12 @@ static Bom copyWithSnapshot(RepositorySystem repositorySystem, Bom bom)
     RepositorySystemSession session = RepositoryUtility.newSession(repositorySystem);
 
     for (Artifact managedDependency : bom.getManagedDependencies()) {
-      String snapshotVersion = findSnapshotVersion(repositorySystem, session, managedDependency);
-      if (snapshotVersion == null) {
-        managedDependencies.add(managedDependency);
-      } else {
-        managedDependencies.add(managedDependency.setVersion(snapshotVersion));
+      Optional<String> snapshotVersion =
```

Comment:
The nice thing about `Optional` here, and the main reason I suggested it, is it lets you avoid this local variable entirely.

```java
findSnapshotVersion(repositorySystem, session, managedDependency)
    .ifPresent(managedDependency::setVersion);
```

What you're doing here is fine, but not very different from what you had before using `null`.

Reply:
Updated:

```
      managedDependencies.add(
          findSnapshotVersion(repositorySystem, session, managedDependency)
              .map(managedDependency::setVersion)
              .orElse(managedDependency));

```
https://github.com/GoogleCloudPlatform/cloud-opensource-java/pull/710/files#diff-df70bea43d3a25e2c723b466bb150234R117